phoenicis-tools/src/main/java/org/phoenicis/tools/system/opener/OpenerProcessImplementation.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (C) 2015-2017 PÂRIS Quentin
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+package org.phoenicis.tools.system.opener;
+
+import java.io.IOException;
+import java.net.URI;
+import java.util.Arrays;
+
+public class OpenerProcessImplementation implements Opener {
+    private final String commandName;
+
+    public OpenerProcessImplementation(String commandName) {
+        this.commandName = commandName;
+    }
+
+    @Override
+    public void open(String file) {
+        final ProcessBuilder processBuilder = new ProcessBuilder(Arrays.asList(commandName, file));
+        try {
+            processBuilder.start().waitFor();
+        } catch (InterruptedException | IOException e) {
+            throw new IllegalStateException(e);
+        }
+    }
+
+    @Override
+    public void open(URI url) {
```

Comment:
The command xdg-open can be used to open a file (a document) or a URL. This is why I want to separate the two behaviors